core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -58,11 +63,148 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
 
   @Before
   override def setUp() {
-    super.setUp
+    super.setUp()
     client = AdminClient.createSimplePlaintext(this.brokerList)
     TestUtils.createTopic(this.zkUtils, topic, 2, serverCount, this.servers)
   }
 
+  @After
+  override def tearDown() {
+    client.close()
+    super.tearDown()
+  }
+
+  @Test
+  def testPurgeDataBeforeWithSeekToBeginning() {
+    val consumer = consumers.head
+
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
```

Comment:
Do we want to poll with 0 since it can burn CPU unnecessarily? Also, instead of duplicating the code for waiting the consumer to be ready, could we factor it out to a private method and reuse?